### PR TITLE
[AIRFLOW-3987] Unify GCP's Connection IDs

### PR DESCRIPTION
**Pull request by ryanyuan**
Unify GCP's Connection IDs

Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow-3987](https://issues.apache.org/jira/browse/AIRFLOW-3987) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Review by potiuk — APPROVED** [review]
Superb!

**Comment by ryanyuan** [comment]
PTAL @kaxil

**Comment by kaxil** [comment]
I'll have a look at this tonight or at most by the end of this week :)

**Comment by Fokko** [comment]
@ryanyuan Could you resolve the conflicts? :-)

File: UPDATING.md [hunk 1]
```diff
@@ -24,6 +24,15 @@ assists users migrating to a new version.
 
 ## Airflow Master
 
+#### Unify default conn_id for Google Cloud Platform
+
+Previously not all hooks and operators related to Google Cloud Platform use
+``google_cloud_default`` as a default conn_id. There is currently one default
+variant. Values like ``google_cloud_storage_default``, ``bigquery_default``,
+``google_cloud_datastore_default`` have been abandoned. The configuration of
```
Comment (kaxil) [inline UPDATING.md]:
```suggestion
``google_cloud_datastore_default`` have been deprecated. The configuration of
```

File: UPDATING.md [hunk 2]
```diff
@@ -24,6 +24,15 @@ assists users migrating to a new version.
...
+connections in the database have been preserved, but in every place where they
+were used, they should be indicated explicitly.
```
Comment (kaxil) [inline UPDATING.md]: Can we rephrase the last line ?

**Review by kaxil — APPROVED** [review]
Minor nits. LGTM

**Comment by ryanyuan** [comment]
@potiuk @kaxil Thanks for the review.

**Review by kaxil — APPROVED** [review]
LGTM. Thanks @ryanyuan

**Pull request merged**